homeassistant/components/dht/sensor.py

```diff
@@ -160,18 +171,27 @@ def update(self):
 class DHTClient:
     """Get the latest data from the DHT sensor."""
 
-    def __init__(self, adafruit_dht, sensor, pin):
+    def __init__(self, sensor, pin, name):
         """Initialize the sensor."""
-        self.adafruit_dht = adafruit_dht
         self.sensor = sensor
-        self.pin = pin
+        self.pin = getattr(board, pin)
         self.data = {}
+        self.name = name
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Get the latest data the DHT sensor."""
-        humidity, temperature = self.adafruit_dht.read_retry(self.sensor, self.pin)
-        if temperature:
-            self.data[SENSOR_TEMPERATURE] = temperature
-        if humidity:
-            self.data[SENSOR_HUMIDITY] = humidity
+        dht = self.sensor(self.pin)
+        try:
+            humidity = dht.humidity
+            temperature = dht.temperature
+            if temperature:
+                self.data[SENSOR_TEMPERATURE] = temperature
+            if humidity:
```

Comment:
Please move the lines that can not raise to an `else:` block.

Reply:
Sorry about that, i know you asked me before and just totally forgot to push that. Pushed now.